tests/unit/utils/test_utils.py

```diff
@@ -126,6 +127,32 @@ def test_makedirs(repo_dir):
     assert os.path.isdir(path_info.fspath)
 
 
+@pytest.mark.parametrize(
+    "parent",
+    [
+        (os.path.join("path", "to", "")),
+        (os.path.join("path", "to")),
+        (os.path.join("path", "")),
+        (os.path.join("path")),
+    ],
+)
+def test_is_child_path_positive(parent):
+    child = os.path.join("path", "to", "folder")
+    assert is_child_path(child, parent)
+
+
+def test_is_child_on_same_path():
+    path = os.path.join("path", "to", "folder")
+    assert not is_child_path(child=path, parent=path)
+
+
+def test_is_child_on_different_path():
+    path1 = os.path.join("path", "to", "folder1")
+    path2 = os.path.join("path", "to", "folder2")
+
+    assert not is_child_path(path1, path2)
```

Comment:
What about `is_child_path('a/b/', 'a/b')`? Comparing only things built with `os.path.join()` doesn't quite handle it.

Reply:
@Suor I think that `os.path.join(path, "")` which is inside `is_child_path` handles this one.

Reply:
Yes, I forgot to test about the case (which fails atm). I'll add that (using `os.path.join()`, as I am not sure about testing via *raw* strings.)

But, is supporting a self-generated (not via `Path`-like or through `os.path.join()`) string a requirement? Because that'd not be platform agnostic. For example, this specific string test might need skipping on `Windows` (to add to that, the implementation is already quite dependent on `os.path.join()` although it claims it's purely lexical :) ).
Shouldn't `is_child_path(r'C:\Users\Documents', r'C:\Users')` also be tested then?

Also, should there be a test for demonstrating support `PathInfo`? (saw #2137)

Reply:
@Suor, I refactored for `PathInfo`, as well as added few tests for absolute path and the case you just mentioned.